pkg/command/nerdctl.go

```diff
@@ -0,0 +1,13 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package command
+
+import "github.com/runfinch/finch/pkg/system"
+
+// NerdctlCommandSystemDeps contains the system dependencies for newNerdctlCommand.
+//
+//go:generate mockgen -copyright_file=../../copyright_header -destination=../../pkg/mocks/nerdctl_cmd_system_deps.go -package=mocks -mock_names NerdctlCommandSystemDeps=NerdctlCommandSystemDeps . NerdctlCommandSystemDeps
+type NerdctlCommandSystemDeps interface {
```

Comment:
Maybe we can move the type definition to `cmd/finch/nerdctl.go` (i.e., where it's used) ([example](https://github.com/runfinch/finch/blob/ff8a35f4ec238abeafb8a1ea7050e844b1996fba/pkg/path/finch.go#L77))

Reply:
that was my original idea, but `go generate` + `mockgen` errors out if operating on non-importable (`package main`) files; e.g. outputs an error about not working with programs, only modules/libraries. If I misunderstood the `go generate`/mockgen errors and someone knows the trick, let me know.

Reply:
We can use the [source mode](https://github.com/golang/mock#source-mode) of `mockgen`.

After moving it to `cmd/finch/nerdctl.go`, we should only need to update the `go generate` comment to be:

```go
mockgen -copyright_file=../../copyright_header -destination=../../pkg/mocks/nerdctl_cmd_system_deps.go -package=mocks -mock_names NerdctlCommandSystemDeps=NerdctlCommandSystemDeps -source=nerdctl.go NerdctlCommandSystemDeps
```

To be specific, `.` is replaced with `-source=nerdctl.go`. Tested locally, and please let me know if it does not work for you, thanks!

Reply:
Thanks! Updated